### PR TITLE
Add a Javascript Channel to sendpostmessage() function within PostMessage.JS

### PR DESCRIPTION
**Pull request by colfal**
Added a javascript channel with the name JSChannel to the .postMessage() method in the sendPostMessage() function.

This allows javascript channel with the name "JSChannel" to be opened in flutter webview to listen for messages

### CODE REVIEW

**Comment by sophtron-jishun** [comment]
handling window messages is tricky, the behavior varies a lot in different environments. 

so the generic window.parent was used to represent the most common case. 

the original code state (before pr #44 ) was to prevent duplicated messages, when the widget detects itself not as a child of another (window.parent === window) it writes to console instead, in which case, no expected parties will be able to receive the message ,  writing to console only was for testing purpose

I didn't think the #44 change was going to fix anything but it should not break anything either, but this change is would be more client specific, correct me if I'm wrong, it seems the object `JSChannel` is only for using flutter webview, which is reasonable but not always the case, looks this change would break the PC browser use, it's would be better to do more local (instead of trying on live) testing with some example html/js code to find out about more scenarios such as android, pc browsers ios etc and add more careful condition checks

File: application/src/utils/PostMessage.js [hunk 1]
```diff
@@ -158,7 +158,9 @@ export function sendPostMessage(event, data, scheme = 'mx') {
       if(window === window.parent){
         console.log('Post message: ' + postUrl);
         console.log(message)
-        window.parent.postMessage(message, postUrl)
+        if(window.JSChannel){
```
Comment (sophtron-jishun) [inline application/src/utils/PostMessage.js:161]: I see a condition is added, thanks!
although, I think it should be more effective to move this condition upper a level and not restricted to line 158

**Comment by colfal** [comment]
I see, and certainly understand your concerns.

Thus far we have been unable to intercept the window event with a messageReceived handler or an eventListener. 

However, we have been seeing the console.logs on 159 & 160, which is why I had hoped adding another window.parent.postMessage would help fix the problem.

The only way I've been able to trigger my message received handler on the flutter side is to postMessage directly through my named Javascript channel like so.
```

controller = webViewController()
..addJavascriptChannel('JSChannel', onMessageReceived (message) {
print("here is your message: ${message.message}")
})

controller.runJavaScript(
'JSChannel.postMessage("this is the only way I can receive a message"));
```

Previously, in the event that if (window === window.parent) evaluated to true on line 158, there was only console logs and no postmessage call.

**Comment by colfal** [comment]
image


Here is an example of what I mean. I can inject my own postMessage to the channel and receive the results on line 47 and lines 33-35, but for whatever reason my eventListener on line 46 to listen for the actual events fired on the sophtron JS side has no effect.

You can see at the bottom of the terminal where I am receiving the console.logs from line 159 & 160 in PostMessage.js your side

**Comment by colfal** [comment]
An alternative solution we are considering here is to replace the `window.parent.postMessage(message, postUrl)` added in PR 44 with 

`window.parent.postMessage(message, "*")`

**Comment by colfal** [comment]
We were fortunately able to find a work around for this, making this PR redundant and unnecessary. Thanks for the support on troubleshooting!

**Pull request closed without merging**